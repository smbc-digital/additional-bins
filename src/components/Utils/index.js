import 'isomorphic-fetch'
import Timeout from 'await-timeout'

async function submitForm(context) {
    const convertedContext = {}
    const timeout = new Timeout()
    
    Object.keys(context).map(key => {
        if (context[key] !== undefined && context[key].value !== undefined) {
            convertedContext[key] = context[key].value
        }
    })

    var doneConvertedContext = JSON.stringify(convertedContext)
    try {
        const result = await Promise.race([
            await fetch('/additional-black-bin/submit', {
                method: 'POST',
                headers: {
                    'Accept': 'application/json; charset=utf-8',
                    'Content-Type': 'application/json; charset=utf-8'
                },
                body: doneConvertedContext
            }),
            timeout.set(10000, 'Timeout!')
        ])

        const paymentUrl = await result.text()
        return {
            status: result.status,
            url: paymentUrl
        }
    }
    catch (error) {
        return {
            status: 400
        }
    }
}

export default submitForm