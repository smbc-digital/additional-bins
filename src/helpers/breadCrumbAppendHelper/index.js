const appendBreadcrumb = (breadcrumbText, url) => {
    let breadcrumbs = document.getElementsByTagName('ul')    
    if(breadcrumbs.length !== 0){
        let ul = breadcrumbs[0]
        let li = document.createElement('li')
        li.innerHTML = '<a href="'+ url+ '">'+ breadcrumbText+'</a>'
        let breadCrumbListItems = ul.getElementsByTagName('li')
        let listSize = breadCrumbListItems.length
        let lastbutone = listSize - 1
        let arr = []
        if (breadCrumbListItems.length !== 0){
           for(let index = 0; index < breadCrumbListItems.length; index++)
           {
                let listItem = breadCrumbListItems[index]
                arr.push(listItem)
           }
           arr.splice(lastbutone, 0, li)           
        }
        ul.innerHTML = ''
        for(let i = 0; i < arr.length; i++){
            ul.appendChild(arr[i])
        }
        
    }
}

export default appendBreadcrumb