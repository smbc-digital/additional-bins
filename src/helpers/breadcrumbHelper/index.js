const showBreadcrumbs = (show) =>{
    let breadcrumb = document.getElementById('breadcrumb')
    if(breadcrumb !== null) 
    {
        if(show){
            breadcrumb.classList.remove('hidden')
        }
        else{
            breadcrumb.classList.add('hidden')
        }
    }
}

const appendBreadCrumb = (breadcrumbText, url) => {
    let breadcrumbs = document.getElementById('breadcrumb')
    if(breadcrumbs !== null){
        var li = document.createElement("li")
        var a = document.createElement("a");
        var text = document.createTextNode(breadcrumbText)
        a.appendChild(text)
        a.setAttribute("href", url)
        li.appendChild(a)
        breadcrumbs.appendChild(li)
}
}

export default showBreadcrumbs