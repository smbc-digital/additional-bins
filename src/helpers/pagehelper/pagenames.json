[
	{
		"number": 1,
		"route": "/additional-black-bin/do-you-recycle"
	},
	{
		"number": 2,
		"route":"/additional-black-bin/need-more-space"
	},
	{
		"number": 3,
		"route":"/additional-black-bin/tell-us-about-yourself"
	},
	{
		"number": 4,
		"route":"/additional-black-bin/payment"
	},
	{
		"number": 6,
		"route":"/additional-black-bin/success"
	},
	{
		"number": 7,
		"route":"/additional-black-bin/how-to-recycle-more"
	},
	{
		"number": 8,
		"route":"/additional-black-bin/order-recycling-bins"
	},
	{
		"number": 11,
		"route":"/additional-black-bin/something-went-wrong"
	}
]