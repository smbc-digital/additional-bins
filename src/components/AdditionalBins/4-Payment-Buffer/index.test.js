import React from 'react'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { getPageRoute } from '../../../helpers/pagehelper'
import * as submitForm from '../../Utils'
import PaymentBuffer from './index'

Enzyme.configure({ adapter: new Adapter() })

describe('PaymentBuffer', () => {
    
    beforeEach(() => {
        fetch.resetMocks()
    })

    it('should call push on submit with the correct next page url when response is 200', async () => {
        // Arrange
        const url = 'http://www.test.url/paymentTest'
        const data = {}
        const history = { push: jest.fn() }
        window.location.assign = jest.fn()
        submitForm.default = jest.fn().mockReturnValue({ status: 200, url: url })

        const wrapper = mount(<PaymentBuffer context={data} history={history} />)
        
       // Act
		await wrapper.find('form').simulate('submit')

		// Assert
        expect(window.location.assign).toHaveBeenCalledWith(url)
    })

    it('should call push on submit with an error page when reponse is not 200', async () => {
        // Arrange
        const url = 'http://www.test.url/paymentTest'
        const data = {}
        const history = { push: jest.fn() }
        window.location.assign = jest.fn()
        submitForm.default = jest.fn().mockReturnValue({ status: 400, url: url })

        const wrapper = mount(<PaymentBuffer context={data} history={history} />)
        
       // Act
		await wrapper.find('form').simulate('submit')

        // Assert
        expect(history.push).toHaveBeenCalledWith(getPageRoute(11))
    })

})