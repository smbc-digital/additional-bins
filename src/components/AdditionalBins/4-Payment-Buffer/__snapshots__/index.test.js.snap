// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentBuffer snapshot renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Request an additional black bin
  </h1>
  <p>
    You're now ready to pay for the bin, use the button below to go to our payments page.
  </p>
  <p>
    Continue to our payment page to fill in your card details and complete your order.
  </p>
  <p>
    The cost is 
    Â£10
    .
  </p>
  <button
    className="button-primary"
    disabled={false}
    id="continue-to-payment"
    onClick={undefined}
    type="submit"
  >
    Continue to payment
  </button>
  <p>
    <a
      href="#"
      onClick={[Function]}
    >
      Previous
    </a>
  </p>
</form>
`;
